src/components/NumberInput/index.js

```diff
@@ -0,0 +1,212 @@
+import React, { useState, useRef } from 'react';
+import styled from 'styled-components';
+import numberDown from '../../assets/properties/displayProperties/icons/numberDown.png';
+
+function NumberInput(props) {
+  const { waitingTime, handleWaitingTime } = props;
+  const [inputValue, setInputValue] = useState(waitingTime);
+  const [upMouseClickStyle, setUpMouseClickStyle] = useState({});
+  const [downMouseClickStyle, setDownMouseClickStyle] = useState({});
+
+  const inputRef = useRef(null);
+
+  const handleValueStepper = action => {
+    const nextValue = +inputValue + action;
+    checkAndSetInput(nextValue);
+    setTimeout(() => {
+      inputRef.current.select();
+    }, 0);
+  };
+
+  const handleKeyDown = e => {
+    if (e.key === 'ArrowDown') {
+      setDownMouseClickStyle({
+        boxShadow: 'inset 2px 4px 4px 1px rgb(85 108 165 / 56%)',
+      });
+    }
+    if (e.key === 'ArrowUp') {
+      setUpMouseClickStyle({
+        boxShadow: 'inset -2px -4px 4px 1px rgb(85 108 165 / 56%)',
+      });
+    }
+  };
+
+  const handleKeyUp = () => {
+    setUpMouseClickStyle({});
+    setDownMouseClickStyle({});
+  };
+
+  const checkAndSetInput = value => {
+    if (!(value < 1 || value > 9999)) {
```

Comment:
1 & 9999 shouldn't be hardcoded, ideally this should support `min` and `max` props

Reply:
max & min props seems a bit like overkill to me, since it's a static condition. And 'max' and 'min' attributes on the input element are not working real time.